### PR TITLE
Add meson flag for compiling with SDL3

### PR DESCRIPTION
**Pull request by ankith26**
Add a flag to allow compilation with SDL3. But for obvious reasons we still default to compiling with SDL2.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -106,47 +111,47 @@ if plat == 'win' and host_machine.cpu_family().startswith('x86')
...
     sdl_lib_dir = sdl_dir / 'lib' / arch_suffix
     pg_inc_dirs += fs.relative_to(sdl_dir / 'include', base_dir)
     pg_lib_dirs += sdl_lib_dir
-    dlls += sdl_lib_dir / 'SDL2.dll'
+    dlls += sdl_lib_dir / '@0@.dll'.format(sdl)
 
-    # SDL2_image
-    sdl_image_dir = prebuilt_dir / 'SDL2_image-@0@'.format(sdl_image_ver)
+    # SDL_image
+    sdl_image_dir = prebuilt_dir / '@0@-@1@'.format(sdl, sdl_image_ver)
```
Comment (MyreMylar) [inline meson.build:122]:
```suggestion
    sdl_image_dir = prebuilt_dir / '@0@-@1@'.format(sdl_image, sdl_image_ver)
```

**Review by MyreMylar — APPROVED** [review]
OK, looks good! 👍

**Comment by ankith26** [comment]
Converting to draft because I want to do a bit more in this PR

**Comment by ankith26** [comment]
- Added a CI for testing that we can build against SDL3
- Add SDL3 guards on meson code. The purpose of this is for incremental porting: we can port modules one by one and keep removing the SDL3 guards as we go
- Fixed SDL3 include. `#include <SDL.h>` no longer works (ref: https://github.com/libsdl-org/SDL/issues/11074)

**Comment by ankith26** [comment]
This PR is now done from my end

**Comment by Starbuck5** [comment]
With regards to the include changes, in my development I've just made sure the include path goes inside the SDL folder instead of editing every include.

**Comment by Starbuck5** [comment]
Another thing I did in my branch to make this stuff compile better was `-DSDL_ENABLE_OLD_NAMES=1`, which is very handy.

Also because of my prior PRs several modules you've masked out in this compile, at least last time I checked (may have changed). Like key, image, base, etc.

**Comment by ankith26** [comment]
The latest changes have broken some of your patches. For instance, `PG_SURF_BitsPerPixel`/`PG_SURF_BytesPerPixel` is now not implementable without an extra function call and a null check.
When I tested this branch locally none of the modules could compile. But I wasn't aware of `-DSDL_ENABLE_OLD_NAMES=1`.

> With regards to the include changes, in my development I've just made sure the include path goes inside the SDL folder instead of editing every include.

That was my first idea but it is virtually impossible to do from our end in a cross platform way (see my older comment with the link to the SDL issue).

**Comment by Starbuck5** [comment]
> The latest changes have broken some of your patches. For instance, PG_SURF_BitsPerPixel/PG_SURF_BytesPerPixel is now not implementable without an extra function call and a null check.

My first reaction: oh boy 😢 

However-- We can use macros on the pixelformat itself, `SDL_BYTESPERPIXEL` and `SDL_BITSPERPIXEL`. Whew.

> That was my first idea but it is virtually impossible to do from our end in a cross platform way (see my older comment with the link to the SDL issue).

I don't know anything about .pc files.

It makes me annoyed at meson that we don't have the freedom to do that. It seems like we should be able to add an include path to an existing dependency object, which would solve this because we could add an include path which is the old include path + "/sdl3".

**Comment by Starbuck5** [comment]
It's so easy on Windows with the old SETUP file that I didn't even bat an eye when I rewrote the include path.

**Review by Starbuck5 — APPROVED** [review]

**Pull request merged**